support/urlUtils.ts

```diff
@@ -167,65 +166,58 @@ export function parseMarker(marker: string): IPromise<Graphic> {
     return promiseUtils.reject();
   }
 
-  return requireUtils
-    .when(require, [
-      "esri/Graphic",
-      "esri/PopupTemplate",
-      "esri/symbols/PictureMarkerSymbol",
-      "esri/symbols/SimpleMarkerSymbol"
-    ])
-    .then(modules => {
-      const [
-        Graphic,
-        PopupTemplate,
-        PictureMarkerSymbol,
-        SimpleMarkerSymbol
-      ] = modules;
-
-      const x = parseFloat(markerArray[0]);
-      const y = parseFloat(markerArray[1]);
-      const content = markerArray[3];
-      const icon_url = markerArray[4];
-      const label = markerArray[5];
-      const wkid = markerArray[2] ? parseInt(markerArray[2], 10) : 4326;
-
-      const markerSymbol = icon_url
-        ? new PictureMarkerSymbol({
-          url: icon_url,
-          height: "32px",
-          width: "32px"
-        })
-        : new SimpleMarkerSymbol({
-          outline: {
-            width: 1
-          },
-          size: 14,
-          color: [255, 255, 255, 0]
-        });
-
-      const point = new Point({
-        x: x,
-        y: y,
-        spatialReference: {
-          wkid: wkid
-        }
-      });
-
-      const hasPopupDetails = content || label;
-      const popupTemplate = hasPopupDetails
-        ? new PopupTemplate({
-          title: content || null,
-          content: label || null
-        })
-        : null;
-
-      const graphic = new Graphic({
-        geometry: point,
-        symbol: markerSymbol,
-        popupTemplate: popupTemplate
-      });
-      return graphic;
-    });
+
+  const [Graphic, PopupTemplate, PictureMarkerSymbol, SimpleMarkerSymbol] = await Promise.all([
+    import("esri/Graphic"),
+    import("esri/PopupTemplate"),
+    import("esri/symbols/PictureMarkerSymbol"),
+    import("esri/symbols/SimpleMarkerSymbol")
+  ]);
+
+
+  const x = parseFloat(markerArray[0]);
+  const y = parseFloat(markerArray[1]);
+  const content = markerArray[3];
+  const icon_url = markerArray[4];
+  const label = markerArray[5];
+  const wkid = markerArray[2] ? parseInt(markerArray[2], 10) : 4326;
+
+  const markerSymbol = icon_url
+    ? new PictureMarkerSymbol.default({
```

Comment:
Do we have to call `.default()` here? Maybe something to look into.

Reply:
I think we do have to - I will double-check with Rene

Reply:
I looked this one up and I think its because we use exports = xxx . There's some info here: https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html under dynamic imports